File: CSESProblemSet/Introductory.Problems/python/MissingNumber.py
class MissingNumber:
    def getMissingNumber(self, numbers, length):
        n = length
        return (n*(n+1)//2) - sum(numbers)   
       
    
    def Solve(self):
        n = int(input())
        numbers = list(map(int, input().split()))
        print(self.getMissingNumber(numbers,n))

if __name__ == '__main__':
    missingNumber = MissingNumber()
    missingNumber.Solve()

File: CSESProblemSet/Introductory.Problems/MissingNumber.cs

/*
You are given all numbers between 1,2,…,n except one. Your task is to find the missing number.
Input
The first input line contains an integer n.
The second line contains n−1 numbers. Each number is distinct and between 1 and n (inclusive).
Output
Print the missing number.
Constraints
•	2≤n≤2⋅10^5
Example
Input: 
5
2 3 1 5
Output: 4

*/
public class MissingNumber
{
    public static int Solve(int n, int[] input)
    {
        Console.WriteLine("Input : " + Array.ConvertAll(input, x => x.ToString()).Aggregate((x, y) => x + " " + y));
        int sum = input.Aggregate((x, y) => x + y);
        return (n * (n + 1) / 2) - sum;
    }

}